Mage.Server/src/main/java/mage/server/MailClient.java

```diff
@@ -1,25 +1,32 @@
 package mage.server;
 
-import java.util.Properties;
+import mage.server.managers.IConfigSettings;
+import mage.server.managers.IMailClient;
+import org.apache.log4j.Logger;
+
 import javax.mail.Message;
 import javax.mail.MessagingException;
 import javax.mail.Session;
 import javax.mail.Transport;
 import javax.mail.internet.InternetAddress;
 import javax.mail.internet.MimeMessage;
-import mage.server.util.ConfigSettings;
-import org.apache.log4j.Logger;
+import java.util.Properties;
 
-public final class MailClient {
+public class MailClient implements IMailClient {
```

Comment:
Why you don't use standard xmage code interfaces notation? Example:
* MailClient -- interface
* MailClient**Impl** -- implementation

Reply:
I'll push the changes with that solution, although I find both of them unsatisfying. By experience, I've noticed that when you are forced to use the `I<Name>` or `<Name>Impl` convention it means that you have not modelled the problem appropriately. In this case: I'm using an interface only to break circular dependencies rather than to describe a contract of a certain functionality, which is fine, but reveals a deeper problem.